### PR TITLE
Create Rich Text Display Component

### LINKED ISSUE #130: Display rich text in Admin "Review and Post"
### Acceptance Criteria
* Build a reusable `RichTextDisplay` component using Draft.js (refer to https://github.com/uwblueprint/pregnancy-centre/pull/122)
* Use this above component on the review and post psage for the role description field

### Technical Notes
https://draftjs.org/

### PR DESCRIPTION
**Pull request by Boyazhang1**
## Ticket link

Closes #130 

## Implementation description
* Created a rich text display component to be used in admin "review and "post"
Screen Shot 2022-03-19 at 11 42 08 PM

## Steps to test
1. Seed the database (run "docker exec -it sistering_backend bash -c "npx prisma db seed")
2. See richTextSamples in comments in CreatePostingReview.tsx
3. Set the child of the RichTextDisplay component used on line 71 of CreatePostingReview.tsx to the different richTextSamples
4. Navigate to http://localhost:3000/admin/posting/create/review to see the richText displayed 

Expected Results (what Albert sees when making the richText components): 

richTextSample1
richtextsample1

richTextSample2
richtextsample2

richTextSample3
richtextsample3

richTextSample4
Screen Shot 2022-03-20 at 1 20 58 PM

## What should reviewers focus on?
* RichTextDisplay component works properly 

## Checklist
- [ x] My PR name is descriptive and in imperative tense
- [ x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [ x] I have run the appropriate linter(s)
- [x ] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

**Review by sherryhli — CHANGES_REQUESTED** [review]
Very cool 😎

There are a couple of lint warnings, please:
1. Remove any unused imports
2. Run `yarn fix` in the frontend folder
3. Remove the rich text test data before merging

Please also install the [draft-js types package](https://www.npmjs.com/package/@types/draft-js) and commit the updated package.json and yarn.lock files
```bash
# if you have yarn locally
cd frontend
yarn add @types/draft-js --dev

# otherwise
docker exec -it sistering_frontend /bin/bash -c "yarn add @types/draft-js --dev"
```

File: frontend/src/components/admin/posting/RichTextDisplay.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    if (children) {
+      setEditorState(
+        EditorState.createWithContent(convertFromRaw(JSON.parse(children))),
+      );
+    }
+  }, [children]);
+
+  return (
+    <Box>
+      <Editor editorState={editorState} readOnly onChange={() => {}} />
```
[the remark this answers is not in the document]
Comment (Boyazhang1) [reply frontend/src/components/admin/posting/RichTextDisplay.tsx:25]: Yep it is a required prop: https://draftjs.org/docs/api-reference-editor/#onchange

File: frontend/src/components/admin/posting/RichTextDisplay.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  children,
+}: RichTextDisplayProps): React.ReactElement => {
+  const [editorState, setEditorState] = useState(EditorState.createEmpty());
+
+  useEffect(() => {
+    if (children) {
+      setEditorState(
+        EditorState.createWithContent(convertFromRaw(JSON.parse(children))),
+      );
+    }
```
[the remark this answers is not in the document]
Comment (Boyazhang1) [reply frontend/src/components/admin/posting/RichTextDisplay.tsx:20]: good question, hmm I guess if the children is falsey, the editorState would just take on the default state which is empty already (EditorState.createEmpty()), so there is no need to set it to empty again ?

File: frontend/src/components/common/RichText/RichTextDisplay.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  const [editorState, setEditorState] = useState<EditorState>(
+    EditorState.createEmpty(),
+  );
+
+  useEffect(() => {
+    if (children) {
+      setEditorState(
+        EditorState.createWithContent(convertFromRaw(JSON.parse(children))),
+      );
+    }
```
Comment (sherryhli) [inline frontend/src/components/common/RichText/RichTextDisplay.tsx:22]: Let's make this part a little more robust, addressing the general case of `children` being an invalid JSON string. Note that we don't have a standardized method for logging frontend errors at the moment, so we'll just use `console.log` for now.

```ts
try {
  const contentState: RawDraftContentState = JSON.parse(children);
  setEditorState(
    EditorState.createWithContent(convertFromRaw(contentState)),
  );
} catch (e: SyntaxError) {
  /* eslint-disable-next-line no-console */
  console.log(`Invalid content state ${children} passed to RichTextDisplay`);
  setEditorState(EditorState.createEmpty());
}
```

**Review by sherryhli — CHANGES_REQUESTED** [review]
Just one last thing! I also see some lint warnings so please run `yarn fix` again 🙂

File: frontend/src/components/admin/posting/RichTextDisplay.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (sherryhli) [reply frontend/src/components/admin/posting/RichTextDisplay.tsx]: Please see new suggestion: https://github.com/uwblueprint/sistering/pull/224#discussion_r834478758

File: frontend/src/components/common/RichText/RichTextDisplay.tsx (hunk 3 above)
Comment (Boyazhang1) [reply frontend/src/components/common/RichText/RichTextDisplay.tsx:22]: Sounds good, it works as expected. I had to change the SyntaxError type to unknown due to the error "Catch clause variable type annotation must be 'any' or 'unknown' if specified".

Error: 
Screen Shot 2022-03-24 at 5 28 50 PM

File: frontend/src/components/common/RichText/RichTextDisplay.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+const RichTextDisplay = ({
+  children,
+}: RichTextDisplayProps): React.ReactElement => {
+  const [editorState, setEditorState] = useState<EditorState>(
+    EditorState.createEmpty(),
+  );
+
+  useEffect(() => {
+    if (children) {
```
Comment (sherryhli) [inline frontend/src/components/common/RichText/RichTextDisplay.tsx:23]: This check shouldn't be needed anymore, a falsey `children` prop will cause `JSON.parse()` to throw so the catch block will handle it

**Review by sherryhli — COMMENTED** [review]
Small nit

**Pull request merged**